components/match2/commons/match_group_input_util.lua

```diff
@@ -891,6 +896,15 @@ function MatchGroupInputUtil.majorityHasBeenWon(bestof, opponents)
 	return false
 end
 
+-- Check if all games/rounds have been played
+---@param playall integer
+---@param opponents {score: integer?}[]
+---@return boolean
+function MatchGroupInputUtil.allHasBeenPlayed(playall, opponents)
+	local scoreSum = Array.reduce(opponents, function(sum, opponent) return sum + (opponent.score or 0) end, 0)
+	return scoreSum >= playall
+end
```

Comment:
```suggestion
function MatchGroupInputUtil.allHasBeenPlayed(games)
	return Array.all(games, function(game) return winner ~= nil end)
end
```

Wouldn't this do it?

Reply:
Hmm you are right in this case then, shouldn't we just refactor `playall` to be a true/false variable, since the number of games for `bestof` is calculated by number of maps that are in the input?

Reply:
not the case on all wikis fwiw

Reply:
Hmm I mean like the use of PlayAll is currently only calculated if PlayAll is non-zero? And thus far this has no representation, for example, in making Ticker use `{{Abbr/PaX}}` instead of `{{Abbr/BoX}}`

Reply:
I like the true/false version personally

Reply:
Ah wait a second, breaks when no maps are provided :/